### PR TITLE
Cherry pick from pr1302 into dev mc 4km jra ryf+regionalpanan

### PR DESCRIPTION
**Pull request by anton-seaice**
This sets the MPI argument `mpi_yield_when_idle` per #1245

This was tested as part of #1288 although setting the MOM parameter NONBLOCKING_UPDATES was found to be not reproducible over restarts (see #1312). So we only set the MPI argument here.

### CODE REVIEW

**Comment by anton-seaice** [comment]
!test repro commit

**Comment by anton-seaice** [comment]
@helenmacdonald @dougiesquire - I think this change make sense for panant?


> adjustments to salt restoring to achieve zero net are made by scaling 
> values without moving the zero contour.

**Comment by dougiesquire** [comment]
I don't think so. This configuration has `ADJUST_NET_SRESTORE_TO_ZERO = False`

**Comment by helenmacdonald** [comment]
Thanks for checking @anton-seaice! I think we might have turned this one off for regional models because when boundary conditions are in play we are not expecting net zero salt flux.

**Comment by anton-seaice** [comment]
!test repro commit

File: config.yaml [hunk 1]
```diff
@@ -75,6 +75,10 @@ collate:
   queue: normalsr
   exe: mppnccombine-fast
 
+mpi:
+  flags:
+  - --mca mpi_yield_when_idle true
+
```
Comment (dougiesquire) [inline config.yaml:77]: There is already an `mpi:` section 13 lines above

**Review by dougiesquire — CHANGES_REQUESTED** [review]

File: config.yaml (hunk 1 above)
Comment (anton-seaice) [reply config.yaml:77]: Ok i've fixed this

**Comment by anton-seaice** [comment]
This config crashed for me at `19000103       40200` , with the same crash as #1251 (different again to https://github.com/ACCESS-NRI/access-om3-configs/issues/1298) !

20260506 120953.284 ERROR            PET0055 No iterations performed in flux_atmocn_mod

20260506 120953.322 ERROR            PET0253 ERROR: 324 nans found in So_u
20260506 120953.323 ERROR            PET0253 ERROR: 324 nans found in So_v

**Comment by dougiesquire** [comment]
We need to get to the bottom of this Nan issue...

**Comment by anton-seaice** [comment]
Ill reconfirm that it also crashes without setting `--mca mpi_yield_when_idle true`

**Comment by anton-seaice** [comment]
Yes - without this change it also crashed at ` atm : model date     19000103       40200`

**Review by dougiesquire — APPROVED** [review]
